pkg/destroy/aws/aws.go

```diff
@@ -504,6 +525,65 @@ func deleteEC2Image(client *ec2.EC2, id string, logger logrus.FieldLogger) error
 		return err
 	}
 
+	logger.Info("Deleted")
+
+	// need to delete the snapshots after deleting the ami. aws will block the
+	// snapshot deletion with an InvalidSnapshot.InUse error until the ami is deleted.
+	return deleteEC2Snapshots(client, snapshots, logger)
+}
+
+func deleteEC2Snapshots(client *ec2.EC2, snapshots []string, logger logrus.FieldLogger) error {
+	failedSnapshotsCh := make(chan string, len(snapshots))
+
+	var wg sync.WaitGroup
+	wg.Add(len(snapshots))
+	for i, id := range snapshots {
+		go func(ix int, snapshotId string) {
+			defer wg.Done()
+			// make at most 10 attempts over around 5 minutes
```

Comment:
I think we want to run forever, until someone kills us (I'm also ok adding a `Context` argument for Go consumers who want to gently cancel).

Reply:
Running forever has the worst-case scenario of blocking the destroy of all of the other resources. But I guess that then the user can run destroy again to clean up the other resources since the subsequent destroys will not attempt to delete the snapshots as they won't be found. I prefer the proposal of tagging the snapshot during destroy over running forever, however.

Reply:
Obsolete by c07b1ed to fcade11.